hack/generator/pkg/codegen/pipeline/verify_no_errored_types.go

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+const VerifyNoErroredTypesStageID = "verifyNoErroredTypes"
+
+// VerifyNoErroredTypes creates a Stage that verifies that no types contain an ErroredType with errors
+func VerifyNoErroredTypes() Stage {
+	return MakeLegacyStage(
+		VerifyNoErroredTypesStageID,
+		"Verify there are no ErroredType's containing errors",
+		func(ctx context.Context, defs astmodel.Types) (astmodel.Types, error) {
+			visitor := newErrorCollectingVisitor()
+
+			for _, def := range defs {
+				_, err := visitor.visitor.Visit(def.Type(), erroredTypeVisitorContext{name: def.Name()})
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed while visiting %q", def.Name())
+				}
+			}
+
+			err := kerrors.NewAggregate(visitor.errs)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
This seems overly complex - why not use an approach similar to the existing `AsObjectType()` and `AsResourceType()` methods? Introduce `AsErroredType()`, use it on `def.Type()` and then access the errors you want right there.

Reply:
Because errors can also be on a `property` in an `ObjectType` or on `spec` or `status` in a `ResourceType`, so you can't just look one level deep to find them. For example today our `Status` error is on `resource.Status`. Unless you're suggesting that we implement a new method on `Type` that's like `Errors()` or something, and then each type can look in its specific viable locations for errors.

Reply:
Ahh, that makes sense. Objections withdrawn.